cache.go

```diff
@@ -0,0 +1,99 @@
+package wazero
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"path"
+	"path/filepath"
+	goruntime "runtime"
+
+	"github.com/tetratelabs/wazero/api"
+	"github.com/tetratelabs/wazero/internal/compilationcache"
+	"github.com/tetratelabs/wazero/internal/version"
+	"github.com/tetratelabs/wazero/internal/wasm"
+)
+
+// Cache is the configuration for caching behavior of wazero.Runtime and can be passed to wazero.RuntimeConfig.
+type Cache interface {
+	// Closer can be used to clear the in-memory cache and this doesn't delete this cache directory when configured.
+	api.Closer
+
+	// WithCompilationCacheDirName configures the destination directory of the compilation cache.
+	// Regardless of the usage of this, the compiled functions are cached in memory, but its lifetime is
+	// bound to the lifetime of wazero.Cache.
+	//
+	// If the dirname doesn't exist, this creates the directory.
+	//
+	// With the given non-empty directory, wazero persists the cache into the directory and that cache
+	// will be used as long as the running wazero version match the version of compilation wazero.
+	//
+	// Note: The embedder must safeguard this directory from external changes.
+	WithCompilationCacheDirName(dir string) error
```

Comment:
maybe simply 
```suggestion
	WithCacheDir(dir string) error
```

Reply:
I don't think the runtime(closeable) object should have a function to change its directory.

We may be ok having `NewCompileCache` and `NewCompileDiskCache` instead?

Reply:
not a fan of the last name, basically it is a write-behind directory...

Reply:
perfect!